web/app/authenticators/torii.ts

```diff
@@ -0,0 +1,28 @@
+// @ts-ignore -- TODO: Add Types
+import Torii from "ember-simple-auth/authenticators/torii";
+import { inject as service } from "@ember/service";
+import FetchService from "hermes/services/fetch";
+
+export default class ToriiAuthenticator extends Torii {
+  @service("fetch") declare fetchSvc: FetchService;
+
+  // Appears unused, but necessary for the session service
+  @service declare torii: unknown;
+
+  async restore(_arguments: unknown) {
+    const data = await super.restore(_arguments);
+    /**
+     * Try the restored credentials with the backend.
+     * If the backend rejects the credentials, the error will bubble up
+     * to the application route's error method, which invalidates the session.
+     */
+    await this.fetchSvc.fetch("/api/v1/me", {
```

Comment:
This is kinda nuanced but I'm not sure if you specifically need to return the promise here - I thought that's how the restore / authenticate hooks worked. If this throws, but you're not returning the rejected promise, I don't know if it will behave properly.

Reply:
@meirish Do you mean that an error here won't bubble up to the application route? (which I _think_ is how this is intended to be handled if I'm understanding the logic correctly)

Reply:
@jfreda You have the right idea. My thought is that this failed call will return a 401 which we let bubble up to application. 

@meirish Are you suggesting that if `restore` doesn't return a promise, we might see unintended consequences?

--

Not to make this more confusing, but, it looks like `restore` has its own rejection handling. I don't think I was right about it bubbling to the error route.

> A rejecting promise indicates that data does not constitute a valid session and will result in the session being invalidated or remaining unauthenticated.

https://ember-simple-auth.com/api/classes/BaseAuthenticator.html#method_restore

Reply:
Yeah the method should return a promise, and on rejection, it kicks you to the route configured for authentication. I’m not sure of the behavior when returning an object or the api call erroring - which are the two cases outlined here.